pkg/upgrade/upgrade.go

```diff
@@ -603,3 +609,43 @@ func GetDeployedRelease(ctx context.Context, cli client.Client) (cluster.Release
 	// could be a clean installation or both CRs are deleted already
 	return cluster.Release{}, nil
 }
+
+func cleanupNimIntegrationTechPreview(ctx context.Context, cli client.Client, oldRelease cluster.Release) error {
+	logger := logf.FromContext(ctx)
+	var errs *multierror.Error
+
+	// TODO where can we get the system namespace, opendatahub | redhat-ods-applications ?
+	ns := "todo-get-namespace"
+
+	if oldRelease.Version.Minor < 16 {
+		cm := &corev1.ConfigMap{}
+		if err := cli.Get(ctx, types.NamespacedName{Name: "nvidia-nim-validation-result", Namespace: ns}, cm); err != nil {
+			if !k8serr.IsNotFound(err) {
+				logger.V(1).Error(err, "failed to fetch tech preview validation result configmap")
```

Comment:
i think it is more useful to tell this is NIM related configmap, instead of tech preview etc.
maybe something "failed to get NIM configmap nvidia-nim-validation-result"

Reply:
same for the error case below

Reply:
Resolved in https://github.com/opendatahub-io/opendatahub-operator/pull/1369/commits/837fee033c89e21d2b7463d1aca7d2ba3813f902.